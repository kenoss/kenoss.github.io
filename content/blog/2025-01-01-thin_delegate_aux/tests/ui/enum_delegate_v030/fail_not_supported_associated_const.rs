use enum_delegate_v030 as enum_delegate;

#[enum_delegate::delegate]
trait ShapeI {
    type Output;
    const NAME: &'static str;

    fn area(&self) -> Self::Output;
}

#[enum_delegate::delegate]
enum Shape {
    Rect(Rect),
    Circle(Circle),
}

struct Rect {
    width: usize,
    height: usize,
}

impl ShapeI for Rect {
    type Output = usize;
    const NAME: &'static str = "Rect";

    fn area(&self) -> Self::Output {
        self.width * self.height
    }
}

struct Circle {
    radius: f64,
}

impl ShapeI for Circle {
    type Output = f64;
    const NAME: &'static str = "Circle";

    fn area(&self) -> Self::Output {
        3.14 * self.radius * self.radius
    }
}

fn main() {}
