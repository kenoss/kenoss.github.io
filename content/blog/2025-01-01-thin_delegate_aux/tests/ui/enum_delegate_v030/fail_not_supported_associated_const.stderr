error: only trait methods with untyped receiver are allowed
 --> tests/ui/enum_delegate_v030/fail_not_supported_associated_const.rs:5:5
  |
5 |     type Output;
  |     ^^^^

error[E0405]: cannot find trait `ShapeI` in this scope
  --> tests/ui/enum_delegate_v030/fail_not_supported_associated_const.rs:22:6
   |
22 | impl ShapeI for Rect {
   |      ^^^^^^ not found in this scope
   |
help: consider importing this trait
   |
1  + use thin_delegate_aux::enum_dispatch::ShapeI;
   |

error[E0405]: cannot find trait `ShapeI` in this scope
  --> tests/ui/enum_delegate_v030/fail_not_supported_associated_const.rs:35:6
   |
35 | impl ShapeI for Circle {
   |      ^^^^^^ not found in this scope
   |
help: consider importing this trait
   |
1  + use thin_delegate_aux::enum_dispatch::ShapeI;
   |
